synapse/lib/storm.py

```diff
@@ -395,11 +400,65 @@ def loadRuntVars(self, query):
                 self.tick()
                 yield node, path
 
-    async def getScopeRuntime(self, query, opts=None):
+    async def getScopeRuntime(self, query, opts=None, impd=False):
+        '''
+        Derive a new runt off of an existing runt. It will pass down any global level vars. It has to
+        re run the oper.run function on any function opers so that it gets the correct context of
+        variable values and functions.
+        '''
         runt = Runtime(self.snap, user=self.user, opts=opts)
+        # imported implies module level
+        runt.isModuleRunt = impd
+        if not impd:  # respect the import boundary
+
+            # if we are a top level module we need to
+            # push down our runtvars
+            if self.isModuleRunt:
+                for name in self.runtvars:
+                    valu = self.vars.get(name, s_common.novalu)
+                    if valu is s_common.novalu:
+                        continue
+                    if name not in self.modulefuncs and name != 'lib':
+                        runt.globals.add(name)
+                        runt.runtvars.add(name)
+                        runt.vars[name] = valu
+
+            # propagate down all the global variables
+            for name in self.globals:
```

Comment:
how is globals different than vars?

Reply:
It's to denote which variables we received from the outer scope, so that we know which ones whose values we can propagate back up and which to avoid any kind of variable pollution.